tests/unit/amo/components/TestCollectionManager.js

```diff
@@ -304,6 +307,24 @@ describe(__filename, () => {
       .toHaveProp('disabled', true);
   });
 
+  it('disables and enables form buttons when modification status changes', () => {
+    const renderAndCheckButtons = (shouldBeDisabled) => {
+      const root = render();
+
+      expect(root.find('.CollectionManager-cancel'))
```

Comment:
Should we be disabling "cancel" at any point? I guess during submit there's the chance the request is posted, we cancel, and get in a weird state forcing a page refresh, but allowing a user to cancel during submission might be nice.

If it's a pain to implement though we could skip it for now.

Reply:
Allowing a user to cancel during submission would be quite complicated, because we'd then have to check whether the API request actually completed successfully, and if it did we'd have to "undo" it. As this is all async, we wouldn't even know if the API request to update the object had been successful when the cancel button is clicked, which makes it even more complicated. The simplest solution, which still protects against multiple actions, is to just disable the whole form (submit and cancel) until we have a result from the API, which is what this patch does.